AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthConfirmSignInTask.swift

```diff
@@ -49,17 +49,32 @@ class AWSAuthConfirmSignInTask: AuthConfirmSignInTask, DefaultLogger {
             AuthPluginErrorConstants.invalidStateError, nil)
 
         guard case .configured(let authNState, _) = await authStateMachine.currentState,
-              case .signingIn(let signInState) = authNState,
-              case .resolvingChallenge(let challengeState, _, _) = signInState else {
+              case .signingIn(let signInState) = authNState else {
             throw invalidStateError
         }
 
-        switch challengeState {
-        case .waitingForAnswer, .error:
-            log.verbose("Sending confirm signIn event: \(challengeState)")
-            await sendConfirmSignInEvent()
-        default:
-            throw invalidStateError
+        if case .resolvingChallenge(let challengeState, let challengeType, _) = signInState {
+
+            // Validate if request valid MFA selection
```

Comment:
updated the comment to be more descriptive. Basically because a string is accepted as a param, there is no way to restrict the users from passing any values (even though we will provide a recommended way). Because of this, we need to validate the users MFA selection to make sure they pass in either SMS or TOTP. 

`confirmSignInMFASelectionResponseError` will contain the messaging for it.